lib/mapping/fast_construct_mapping.cpp

```diff
@@ -75,9 +75,9 @@ void fast_construct_mapping::construct_initial_mapping_topdown_internal( Partiti
                 std::vector< NodeID > & map_to_original, 
                 std::vector< NodeID > & perm_rank) {
 
+        if( group_sizes.size() == 0 ) return;
         PartitionID num_parts = group_sizes[group_sizes.size()-1];
         if( num_parts == 1 ) {
-                if( group_sizes.size() == 1 ) return;
```

Comment:
Hi Carl, why is this changed? Could you restore this to the original? The other changes look fine.
And could you also add yourself to the bottom of the readme.

Thanks
Christian